system/Cache/Handlers/FileHandler.php

```diff
@@ -44,14 +54,23 @@ class FileHandler implements CacheInterface
 	 */
 	public function __construct(Cache $config)
 	{
-		$path = ! empty($config->storePath) ? $config->storePath : WRITEPATH . 'cache';
-		if (! is_really_writable($path))
+		if (isset($config->file))
```

Comment:
If this returns `false`, then `$this->mode` will not be initialized. Either, we add that to the `else` or add a default value to the property `$mode`.

Reply:
there is already `??` for that as fallback

Reply:
If only `isset($config->file)` resolves to true then there's is fallback. If the Cache config is not updated to use the new property, `$mode` is uninitialized.

https://3v4l.org/qFvvV

Reply:
I think we can do with property_exisrts then https://3v4l.org/uFhnG

Reply:
Hmm. I like that solution as it is cleaner.

Reply:
Good catch and elegant solution! I will implement this later on desktop.

Reply:
I had to modify this a bit to keep the backwards-compatible support for `$config->storePath` but I think the spirit of it is there.